code/modules/uplink/uplink_item.dm

```diff
@@ -101,7 +101,10 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 		for(var/obj/item/I in B)
 			U.purchase_log += "<big>\icon[I]</big>"
 	else
-		U.purchase_log += "<big>\icon[A]</big>"
+		// Don't add /obj/item/stack/telecrystal to the purchase_log since
+		// it's just used to buy more items (including itself!)
+		if(!istype(src, /datum/uplink_item/device_tools/telecrystal))
```

Comment:
❄️ 

Make this more OOP and based upon the datum itself rather than a snowflake istype check.